### DISCUSSION TITLE
Practical Limit to Number of Wiki Pages in Wiki.js?

### DISCUSSION THREAD

**Discussion by chendy** [requarks/wiki#5602] [Help / Questions] [answered]
Just wanted to understand the rough number of pages wiki.js can support. Say with a standard high-performance hosted database?

Thanks

**Comment by rrg92** [comment]
I am also interested in knowing this. I plan to create more than 2000 pages/month and wonder if wiki.js can support well this amount.

**Comment by NGPixel** [comment, marked as answer]
The only limit you might hit on Wiki.js side is the rebuild tree job. When creating / updating a page, the table that contains the tree of all pages is completely rebuilt. Right now this process doesn't use query streaming, so you might run out of RAM if you have an insane amount of pages (e.g. millions) and not enough memory. This job could be improved in the future to use streaming and be better optimized.

Other than that, your database will probably be the next thing to be affected, especially if you have DB search enabled.
